storybookjs/storybook
# 6975: @storybook/ui TS incremental migration - Part 1

## What I did

- Turned off `noImplicitAny` flag to make sure the package can be built.
- Break big monolith code into modular functions/files.
- Progressively add type declarations.

## How to test

- Is this testable with Jest or Chromatic screenshots? 
Use the existed tests. Should have nothing effected.

- Does this need a new example in the kitchen sink apps?
No.

- Does this need an update to the documentation?
No.

---

As discussed with @ndelangen, it would be more pragmatic to migrate small chunks of code into TS, meaning it will take several incremental steps. Please also have a look in #6975.

Please note, many typing information is still missed. It is more just a housekeeping task. My goal is to get PR as small as possible so my work won't be drained or blocking any one. It would also important to get calibrated from early feedbacks.  :)
